mission_control/src/mission_control.cpp

```diff
@@ -271,91 +194,94 @@ void MissionControlNode::loadMissionCallback(const mission_control::LoadMission:
     outmsg.load_state = ReportLoadMissionState::FAILED;
     outmsg.mission_id = 0;
-    ROS_INFO("loadMissionCallback - FAILED to parse mission file '%s'",
-             (msg->mission_file_full_path).c_str());
+    ROS_INFO_STREAM("loadMissionCallback - FAILED to parse mission file "
+                    << msg->mission_file_full_path.c_str());
   }
   else
   {
     outmsg.load_state = ReportLoadMissionState::SUCCESS;
     outmsg.mission_id = m_mission_id_counter;
-    ROS_INFO("loadMissionCallback - SUCCESSFULLY parsed mission file '%s'",
-             (msg->mission_file_full_path).c_str());
+    ROS_INFO_STREAM("loadMissionCallback - SUCCESSFULLY parsed mission file "
+                    << msg->mission_file_full_path.c_str());
   }
 
   outmsg.header.stamp = ros::Time::now();
-
   pub_report_mission_load_state.publish(outmsg);
 }
 
 void MissionControlNode::executeMissionCallback(
     const mission_control::ExecuteMission::ConstPtr& msg)
 {
-  // TODO(qna): Need to shutdown any mission that is currently running.
-  ROS_INFO("executeMissionCallback - Executing mission id[%d]", msg->mission_id);
-  executeMission(msg->mission_id);
+  if (m_current_mission_id > 0)
+  {
+    BT::NodeStatus missionStatus = m_mission_map[m_current_mission_id]->getMissionStatus();
+    if ((missionStatus != BT::NodeStatus::IDLE) && (missionStatus != BT::NodeStatus::SUCCESS))
+    {
+      ROS_INFO_STREAM("There is a mission being executed - mission id["
+                      << m_current_mission_id << "] - Mission Status: "
+                      << m_mission_map[m_current_mission_id]->getMissionStatus());
+      return;
+    }
+  }
+
+  if ((msg->mission_id > 0) && (m_mission_map.size() > 0) &&
+      (m_mission_map.count(msg->mission_id) > 0))
+  {
+    m_current_mission_id = msg->mission_id;
+    executeMissionTimer.start();
+    ROS_INFO_STREAM("executeMissionCallback - Executing mission id[" << m_current_mission_id
+                                                                     << "]");
+  }
+  else
+  {
+    ROS_INFO_STREAM("Error - Wrong Mission Number. "
+                    << "Mission Number: " << msg->mission_id);
+  }
 }
 
 void MissionControlNode::abortMissionCallback(const mission_control::AbortMission::ConstPtr& msg)
 {
-  // TODO(qna): Need to shutdown any mission that is currently running.
-  // TODO(qna): should probably also check the timestamp from the message
-  ROS_INFO("abortMissionCallback - Aborting mission id[%d]", msg->mission_id);
-  abortMission(msg->mission_id);  // assume mission ID is the current mission
+  abortMission(msg->mission_id);
+  ROS_INFO_STREAM("abortMissionCallback - Aborting mission id " << msg->mission_id);
 }
 
 void MissionControlNode::queryMissionsCallback(const mission_control::QueryMissions::ConstPtr& msg)
 {
   ReportMissions outmsg;
   MissionData data;
-  std::map<int, Mission*>::iterator it;
-
-  for (it = m_mission_map.begin(); it != m_mission_map.end(); it++)
+  for (const auto& it : m_mission_map)
```

Comment:
@patapufate consider: 
```suggestion
  for (const auto& entry : m_mission_map)
```
`it` suggests iterator, and you're using a range-for.

Reply:
fixed